llvm/lib/Target/X86/X86InstrArithmetic.td

```diff
@@ -1107,43 +1107,85 @@ def : Pat<(store (X86adc_flag GR64:$src, (loadi64 addr:$dst), EFLAGS),
 
 // Patterns for basic arithmetic ops with relocImm for the immediate field.
 multiclass ArithBinOp_RF_relocImm_Pats<SDNode OpNodeFlag, SDNode OpNode> {
```

Comment:
Maybe split it into 2 multiclass, e.g.,
```
multiclass ArithBinOp_RF_relocImm_Pats_All<SDNode OpNodeFlag, SDNode OpNode> {
  defm : ArithBinOp_RF_relocImm_Pats<OpNodeFlag, OpNode, 0>,  ArithBinOp_RF_relocImm_Pats<OpNodeFlag, OpNode, 1>;
}
multiclass ArithBinOp_RF_relocImm_Pats<SDNode OpNodeFlag, SDNode OpNode, bit ndd> {
  let Predicates = !if(!eq(ndd, 0), [NoNDD], [HasNDD]) in {
    def : Pat<(OpNodeFlag GR8:$src1, relocImm8_su:$src2),
              (!cast<Instruction>(NAME#"8ri"#!if(!eq(ndd, 0), "", "_ND")) GR8:$src1, relocImm8_su:$src2)>;
  }
}
```
just add a `bit ndd` argument and then

Reply:
Why do we need to define these patterns again given we defined during defining the instruction?

Reply:
Never mind, I see the comment above.

Reply:
> Maybe split it into 2 multiclass, e.g
I didn't do it b/c mi_ND does not have the "store" operation.